base_location_nuts/wizard/nuts_import.py

```diff
@@ -133,15 +132,16 @@ def _download_nuts(self, url_base=None, url_path=None, url_params=None):
         logger.info('Download successfully %d bytes' %
                     len(res_request.content))
         # Workaround XML: Remove all characters before <?xml
-        pattern = re.compile(r'^.*<\?xml', re.DOTALL)
-        content_fixed = re.sub(pattern, '<?xml', res_request.content)
-        if not re.match(r'<\?xml', content_fixed):
+        pattern = re.compile(r'^.*<\?xml'.encode('utf-8'), re.DOTALL)
+        content_fixed = re.sub(pattern, b'<?xml',
+                               res_request.content)
+        if not re.match(r'<\?xml'.encode('utf-8'), content_fixed):
             raise UserError(_('Downloaded file is not a valid XML file'))
         return content_fixed
 
     @api.model
     def _load_countries(self):
-        for k in self._countries.keys():
+        for k in list(self._countries.keys()):
```

Comment:
Replace with `for k in self._countries:`

Reply:
This is done automatically by `2to3`, and there are reasons for that:

* `self._countries` doesn't return now the keys by default 
* It's converted to list because they don't get if it's going to be used as an iterator. In this case `list(` can be removed.

Reply:
This is a combination of both things, but both things are wrong in this specific cases AFAICS.

Check this out, from last PyCon: https://youtu.be/V5-JH23Vk0I?t=3m17s